src/js/directoryLinksProvider.js

```diff
@@ -0,0 +1,130 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+ /*globals Request, CacheTasks, async, ProviderManager, Links*/
+
+"use strict";
+
+// The const that tells where to obtain directory links
+const PREF_DIRECTORY_SOURCE = "https://tiles.services.mozilla.com/v3/links/fetch/en-US/nightly";
+
+// The frecency of a directory link
+const DIRECTORY_FRECENCY = 1000;
+
+/**
+ * Emits notifications to PlacesProvider and Links
+ */
+const DirectoryLinksProvider = {
+  _globalFetch: fetch,
+
+  _observers: new Set(),
+
+  _directoryLinksRequest: new Request(PREF_DIRECTORY_SOURCE),
```

Comment:
It's probably fine to just use PREF_DIRECTORY_SOURCE instead of creating the Request object. Only create a request object when you need to modify the browser's request in some way (e.g., we need to add a custom request header).